internal/docker/docker.go

```diff
@@ -337,6 +348,101 @@ func (c *Docker) runCommand(cmdName string, args ...string) error {
 	return err
 }
 
+// partialCopyRepo copies the following files from the [config.Config.Repo] to
+// [ReleaseRequest.PartialRepoDir] in the given ReleaseRequest:
+//
+// 1. all directories that make up all libraries, or one library, if the library
+// ID is specified.
+//
+// 2. the .librarian directory.
+//
+// 3. global files declared in config.yaml.
+func partialCopyRepo(request *ReleaseRequest) error {
+	dst := request.PartialRepoDir
+	src := request.Cfg.Repo
+	if err := os.MkdirAll(dst, 0755); err != nil {
+		return fmt.Errorf("failed to make directory: %w", err)
+	}
+
+	for _, library := range request.State.Libraries {
+		// Only copy files that make up one library.
+		if request.LibraryID != "" {
+			if library.ID == request.LibraryID {
+				if err := copyOneLibrary(dst, src, library); err != nil {
+					return err
+				}
+				break
+			}
+			continue
+		}
+
+		// Copy all files make up all libraries.
+		if err := copyOneLibrary(dst, src, library); err != nil {
+			return err
+		}
+	}
+
+	// Copy the .librarian directory.
+	if err := os.CopyFS(
+		filepath.Join(dst, config.LibrarianDir),
+		os.DirFS(filepath.Join(src, config.LibrarianDir))); err != nil {
+		return fmt.Errorf("failed to copy librarian dir to %s: %w", dst, err)
+	}
+
+	// Copy global files declared in global config, skipping write only files.
+	for _, globalFile := range request.GlobalConfig.GlobalFilesAllowlist {
+		if globalFile.Permissions == config.PermissionWriteOnly {
+			continue
+		}
+
+		dstPath := filepath.Join(dst, globalFile.Path)
+		srcPath := filepath.Join(src, globalFile.Path)
+		if err := copyFile(dstPath, srcPath); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func copyOneLibrary(dst, src string, library *config.LibraryState) error {
+	for _, srcRoot := range library.SourceRoots {
+		dstPath := filepath.Join(dst, srcRoot)
+		srcPath := filepath.Join(src, srcRoot)
+		if err := os.CopyFS(dstPath, os.DirFS(srcPath)); err != nil {
+			return fmt.Errorf("failed to copy %s to %s: %w", library.ID, dst, err)
+		}
+	}
+
+	return nil
+}
+
+func copyFile(dst, src string) (err error) {
+	sourceFile, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("failed to open file: %s", src)
```

Comment:
Same goes for the couple below:
```suggestion
		return fmt.Errorf("failed to open file %q: %w", src, err)
```